ide/services/ide-git/src/main/java/org/eclipse/dirigible/runtime/git/service/GitRestService.java

```diff
@@ -621,6 +626,55 @@ public Response removeFileFromIndex(@ApiParam(value = "Name of the Workspace", r
 		return Response.ok().build();
 	}
 
+	@GET
+	@Path("/{project}/origin-urls")
+	@Produces("application/json")
+	@ApiOperation("Get Project Repo URLs")
+	@ApiResponses({ @ApiResponse(code = 200, message = "Git Project Repo URLs") })
+	public Response getOriginUrl(@PathParam("workspace") String workspace, @PathParam("project") String project) throws GitConnectorException {
+		String user = UserFacade.getName();
+		if (user == null) {
+			return createErrorResponseForbidden(NO_LOGGED_IN_USER);
+		}
+
+		ProjectOriginUrls originUrls = processor.getOriginUrls(workspace, project);
+		return Response.ok(originUrls).build();
+	}
+
+	@GET
+	@Path("/{project}/set-fetch-url")
+	@Produces("application/json")
+	@ApiOperation("Set origin fetch URL")
+	@ApiResponses({ @ApiResponse(code = 200, message = "Git File Diff") })
+	public Response setFetchUrl(@PathParam("workspace") String workspace, @PathParam("project") String project, @QueryParam("url") String url) throws GitConnectorException, GitAPIException, URISyntaxException {
+		String user = UserFacade.getName();
+		if (user == null) {
+			return createErrorResponseForbidden(NO_LOGGED_IN_USER);
+		}
+		processor.setFetchUrl(workspace, project, url);
+		JSONObject res = new JSONObject();
+		res.put("status", "success");
+		res.put("url", url);
+		return Response.ok().entity(res).type(ContentTypeHelper.APPLICATION_JSON).build();
+	}
+
+	@GET
+	@Path("/{project}/set-push-url")
+	@Produces("application/json")
+	@ApiOperation("Set origin push URL")
+	@ApiResponses({ @ApiResponse(code = 200, message = "Git File Diff") })
+	public Response setPushUrl(@PathParam("workspace") String workspace, @PathParam("project") String project, @QueryParam("url") String url) throws GitConnectorException, GitAPIException, URISyntaxException {
+		String user = UserFacade.getName();
+		if (user == null) {
+			return createErrorResponseForbidden(NO_LOGGED_IN_USER);
+		}
+		processor.setPushUrl(workspace, project, url);
+		JSONObject res = new JSONObject();
+		res.put("status", "success");
+		res.put("url", url);
+		return Response.ok().entity(res).type(ContentTypeHelper.APPLICATION_JSON).build();
+	}
```

Comment:
I guess the `set` operation should be a done via `@POST` HTTP method?
Having in mind some of the best practices for design a REST API,  the `@Path` annotations should adjusted as follows:
- `@Path("/{project}/set-fetch-url")` - > `@Path("/{project}/fetch-url")`
- `@Path("/{project}/set-push-url")` -> `@Path("/{project}/push-url")`

Reply:
Agree.